src/Microsoft.Data/DataFrame.cs

```diff
@@ -34,6 +35,132 @@ public DataFrame(IList<BaseColumn> columns)
             _table = new DataFrameTable(columns);
         }
 
+        public DataFrame(RecordBatch recordBatch)
+        {
+            _table = new DataFrameTable();
+            Apache.Arrow.Schema arrowSchema = recordBatch.Schema;
+            int fieldIndex = 0;
+            IEnumerable<IArrowArray> arrowArrays = recordBatch.Arrays;
+            foreach (IArrowArray arrowArray in arrowArrays)
+            {
+                Field field = arrowSchema.GetFieldByIndex(fieldIndex);
+                IArrowType fieldType = field.DataType;
+                BaseColumn dataFrameColumn = null;
+                switch (fieldType.TypeId)
+                {
+                    case ArrowTypeId.Boolean:
+                        ReadOnlyMemory<byte> valueBuffer = (arrowArray as BooleanArray).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> nullBitMapBuffer = (arrowArray as BooleanArray).NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<bool>(field.Name, valueBuffer, nullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Decimal:
+                        ReadOnlyMemory<byte> decimalValueBuffer = (arrowArray as PrimitiveArray<decimal>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> decimalNullBitMapBuffer = (arrowArray as PrimitiveArray<decimal>).NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<decimal>(field.Name, decimalValueBuffer, decimalNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Double:
+                        ReadOnlyMemory<byte> doubleValueBuffer = (arrowArray as PrimitiveArray<double>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> doubleNullBitMapBuffer = (arrowArray as PrimitiveArray<double>).NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<double>(field.Name, doubleValueBuffer, doubleNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Float:
+                        ReadOnlyMemory<byte> floatValueBuffer = (arrowArray as PrimitiveArray<float>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> floatNullBitMapBuffer = (arrowArray as PrimitiveArray<float>).NullBitmapBuffer.Memory;
+                        dataFrameColumn = new PrimitiveColumn<float>(field.Name, floatValueBuffer, floatNullBitMapBuffer, arrowArray.Length, arrowArray.NullCount);
+                        break;
+                    case ArrowTypeId.Int8:
```

Comment:
Does Int8 and Int16 arrays really cast to `PrimitiveArray<int>`? I don't think this will work.

Do we have a test for it?